GX-TimeFrequency/src/minimax_utils.F90

```diff
@@ -0,0 +1,57 @@
+! ***************************************************************************************************
+!  Copyright (C) 2020-2022 Green-X library                                                          
+!  This file is distributed under the terms of the APACHE2 License.                                 
+!                                                                                                   
+! ***************************************************************************************************
+!> \brief This module contains auxiliary procedures and data structures for the main minimax routines
+! ***************************************************************************************************
+module minimax_utils
+#include "gx_common.h"
+  use kinds, only: dp
+  implicit none
+
+  private
+
+  type :: er_aw_aux
```

Comment:
As this contains allocatables, you might consider adding initialiser and finaliser type-bound procedures, just to handle the memory.

Reply:
Thanks a lot for the comments @AlexBuccheri ! This is a cool idea, I will implement it

Reply:
I will keep as it is. For later developing.